### PR TITLE
[LTC] Replace XLA_FN_COUNTER with TORCH_LAZY_FN_COUNTER

### PR DESCRIPTION
**Pull request by alanwaketan**
Summary:
TORCH_LAZY_FN_COUNTER supports prefix op names with any string. Therefore, we can just reuse it to add xla prefix to xla ops.

This is  paired with pytorch/pytorch#89608.

Test Plan:
CI.

### CODE REVIEW

**Comment by alanwaketan** [comment]
To be noted, this PR depends on #4208. Will need a rebase before landing.

File: scripts/gen_lazy_tensor.py [hunk 1]
```diff
@@ -112,7 +112,7 @@ def build_ir_node(self, func: NativeFunction, schema: LazyIrSchema) -> str:
       get_tensorlist="GetTensorList",
       get_tensor_or_wrap_number="bridge::GetXlaTensorOrCreateForWrappedNumber",
       try_get_tensor="bridge::TryGetXlaTensor",
-      metrics_counter='XLA_FN_COUNTER("xla::")',
+      metrics_counter='TORCH_LAZY_FN_COUNTER("xla::")',
```
Comment (JackCaoG) [inline scripts/gen_lazy_tensor.py:115]: I think `TORCH_LAZY_FN_COUNTER` is default so maybe we don't need to pass it? I am not sure if `xla::` will also be there through.

**Review by JackCaoG — APPROVED** [review]
mostly lgtm, thanks!

File: scripts/gen_lazy_tensor.py (hunk 1 above)
Comment (alanwaketan) [reply scripts/gen_lazy_tensor.py:115]: The default is "lazy::". We need to replace that with "xla::".

**Pull request merged**